sycl/include/sycl/detail/spirv.hpp

```diff
@@ -983,13 +1007,22 @@ EnableIfNativeShuffle<T> ShuffleUp(GroupT g, T x, uint32_t delta) {
                                           convertToOpenCLType(x), delta);
   }
 #else
+  T result;
   if constexpr (ext::oneapi::experimental::is_user_constructed_group_v<
                     GroupT>) {
-    return __nvvm_shfl_sync_up_i32(detail::ExtractMask(detail::GetMask(g))[0],
-                                   x, delta, 0);
+    auto MemberMask = detail::ExtractMask(detail::GetMask(g))[0];
+    if constexpr (is_fixed_size_group_v<GroupT>) {
+      CUDA_SHFL_SYNC(result, MemberMask, x, delta, 0, up_i32)
+    } else {
+      unsigned localSetBit = g.get_local_id()[0] + 1;
+      int unfoldedSrcSetBit = localSetBit - delta;
+      CUDA_SHFL_SYNC(result, MemberMask, x,
+                     __nvvm_fns(MemberMask, 0, unfoldedSrcSetBit), 31, idx_i32)
+    }
   } else {
-    return __nvvm_shfl_sync_up_i32(membermask(), x, delta, 0);
+    CUDA_SHFL_SYNC(result, membermask(), x, delta, 0, up_i32)
   }
+  return result;
 #endif
 }
```

Comment:
We should undefine the macro again, as otherwise we will be leaking it into user code. Alternatively, you can make `CUDA_SHFL_SYNC` into a function and rely on inlining.